flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/call.scala

```diff
@@ -49,6 +57,128 @@ case class Call(functionName: String, args: Seq[Expression]) extends Expression
 }
 
 /**
+  * Over expression for calcite over transform.
+  *
+  * @param agg             over-agg expression
+  * @param aggAlias        agg alias for following `select()` clause.
+  * @param overWindowAlias over window alias
+  * @param overWindow      over window
+  */
+case class OverCall(
+    agg: Aggregation,
+    var aggAlias: Expression,
+    overWindowAlias: Expression,
+    var overWindow: OverWindow = null) extends Expression {
+
+  private[flink] def as(aggAlias: Expression): OverCall = {
```

Comment:
If we remove the `as` method from `OverCall`, it will be handled as a regular`Alias` expression which wraps the `OverCall`. 
Since the `OverCall` is wrapped, the `ProjectTranslator.translateOverWindow()` method needs another `case Alias(OverCall(agg, alias, _), aggAlias, _)` when inserting the OverWindow into the call.